### PR TITLE
Fix improvement and cleanup for micro

### PR DESCRIPTION
**Pull request by LKHN**
* Fixed package counts consistency from 18/102/116 to 18 on all build
environments.
* Cleanup: The image size reduced from 36MB to 35.8MB(x86_64), 50.6MB(aarch64).
* Fixed: Creates missing /run/lock

Signed-off-by: Elkhan Mammadli <elkhan.mammadli@protonmail.com>

### CODE REVIEW

File: kickstarts/almalinux-8-micro.aarch64.ks [hunk 1]
```diff
@@ -75,22 +64,21 @@ echo '%_install_langs en_US.UTF-8' > /etc/rpm/macros.image-language-conf
...
-systemctl mask console-getty.service \
-               dev-hugepages.mount \
-               getty.target \
-               systemd-logind.service \
-               sys-fs-fuse-connections.mount \
-               systemd-remount-fs.service
-
 # remove unnecessary files
-rm -f /var/lib/dnf/history.* \
+rm -rfv /var/lib/dnf \
```
Comment (ezamriy) [inline kickstarts/almalinux-8-micro.aarch64.ks:68]: Why do we need verbosity (`-v`) here?

File: kickstarts/almalinux-8-micro.aarch64.ks [hunk 2]
```diff
@@ -75,22 +64,21 @@ echo '%_install_langs en_US.UTF-8' > /etc/rpm/macros.image-language-conf
...
       /run/nologin
-rm -fr /var/log/* \
+rm -rfv /var/log/* \
```
Comment (ezamriy) [inline kickstarts/almalinux-8-micro.aarch64.ks:70]: Why do we need verbosity (`-v`) here?

File: kickstarts/almalinux-8-micro.aarch64.ks [hunk 3]
```diff
@@ -75,22 +64,21 @@ echo '%_install_langs en_US.UTF-8' > /etc/rpm/macros.image-language-conf
...
        /boot || true
 %end
+
+%post --nochroot --logfile=/mnt/sysimage/root/anaconda-post-nochroot.log --erroronfail
+set -eux
+
+# https://bugzilla.redhat.com/show_bug.cgi?id=1343138
+umount /mnt/sysimage/run
+# We don't have systemd and need to create manually
+chroot /mnt/sysimage install -d /run/lock -m 0755 -o root -g root
```
Comment (ezamriy) [inline kickstarts/almalinux-8-micro.aarch64.ks:81]: Do we really need to chroot for that? root uid/gid will always be the same, so why not just create a `/mnt/sysimage/run/lock` directory similarly as you did on the next line?

File: kickstarts/almalinux-8-micro.aarch64.ks (hunk 1 above)
Comment (LKHN) [reply kickstarts/almalinux-8-micro.aarch64.ks:68]: To see, what is removed in /root/anaconda-post.log file.

File: kickstarts/almalinux-8-micro.aarch64.ks (hunk 3 above)
Comment (srbala) [reply kickstarts/almalinux-8-micro.aarch64.ks:81]: Agree. After these two line changes breaks `ks2rootfs` compatablity. until that available, use old or alternative

File: kickstarts/almalinux-8-micro.x86_64.ks [hunk 4]
```diff
@@ -22,41 +22,30 @@ shutdown
...
--xz
--yum
 %end
 
+%pre
+# Pre configure tasks for Docker
+
+# Don't add the anaconda build logs to the image
+# see /usr/share/anaconda/post-scripts/99-copy-logs.ks
+touch /tmp/NOSAVE_LOGS
```
Comment (srbala) [inline kickstarts/almalinux-8-micro.x86_64.ks:47]: Just a question, is this requried for other types like minimal, init etc?

Comment (ezamriy) [reply kickstarts/almalinux-8-micro.x86_64.ks:47]: I believe yes (btw, nice catch @LKHN). People don't need installer logs in containers and we can reduce image size.

File: kickstarts/almalinux-8-micro.x86_64.ks [hunk 5]
```diff
@@ -75,22 +64,21 @@ echo '%_install_langs en_US.UTF-8' > /etc/rpm/macros.image-language-conf
...
+rm -rfv /var/log/* \
        /tmp/* /tmp/.* \
        /boot || true
 %end
+
+%post --nochroot --logfile=/mnt/sysimage/root/anaconda-post-nochroot.log --erroronfail
+set -eux
+
+# https://bugzilla.redhat.com/show_bug.cgi?id=1343138
+umount /mnt/sysimage/run
```
Comment (srbala) [inline kickstarts/almalinux-8-micro.x86_64.ks:79]: These needs to be hold until its combatable to `ks2rootfs`

**Review by srbala — CHANGES_REQUESTED** [review]
What was the upstream packages count?

**Comment by srbala** [comment]
@LKHN How are you building aarch64 builds?

File: kickstarts/almalinux-8-micro.aarch64.ks (hunk 3 above)
Comment (LKHN) [reply kickstarts/almalinux-8-micro.aarch64.ks:81]: It looks same without chroot.
```
bash-4.4# stat /run
  File: /run
  Size: 18              Blocks: 0          IO Block: 4096   directory
Device: 2fh/47d Inode: 17524580    Links: 3
Access: (0755/drwxr-xr-x)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2021-09-03 10:42:22.828981698 +0000
Modify: 2021-09-03 10:41:50.000000000 +0000
Change: 2021-09-03 10:42:22.822981670 +0000
 Birth: -
bash-4.4# stat /run/lock/
  File: /run/lock/
  Size: 6               Blocks: 0          IO Block: 4096   directory
Device: 2fh/47d Inode: 34884318    Links: 2
Access: (0755/drwxr-xr-x)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2021-09-03 10:42:22.828981698 +0000
Modify: 2021-09-03 10:41:50.000000000 +0000
Change: 2021-09-03 10:42:22.822981670 +0000
 Birth: -
```

Comment (LKHN) [reply kickstarts/almalinux-8-micro.aarch64.ks:81]: @srbala Oh! actual problem is `%post --nochroot` part right? :thinking:

**Comment by LKHN** [comment]
> @LKHN How are you building aarch64 builds?

With first method on AWS Graviton2 based EC2 instance with AlmaLinux AMI.

**Comment by srbala** [comment]
@LKHN @ezamriy Please setup an amazon accout for my dev/testing

**Review by srbala — APPROVED** [review]
This PR and `minimal` ks changes breaks  `almalinux/ks2rootfs` build, hope it would be fixed post release

**Pull request merged**